domain/application/state/types.go

```diff
@@ -673,3 +678,25 @@ type revisionUpdaterApplicationNumUnits struct {
 	UUID     string `db:"uuid"`
 	NumUnits int    `db:"num_units"`
 }
+
+type applicationConfig struct {
+	Key   string `db:"key"`
+	Value any    `db:"value"`
+	Type  string `db:"type"`
+}
+
+type setApplicationConfig struct {
+	ApplicationUUID string `db:"application_uuid"`
+	Key             string `db:"key"`
+	Value           any    `db:"value"`
+	TypeID          int    `db:"type_id"`
```

Comment:
Why not passing the string `Type` as it's returned from `applicationConfig`?

Reply:
Consistency, we're always using the ID, as it allows the DB to change names, whilst keeping the ID.